### PR TITLE
Focus is now trapped when modal is opened. It works in both directions with tab or shift+tab.

### PR DESCRIPTION
**Pull request by paulorbpacheco**
Fix Modal: Focus is now trapped inside when modal is opened. It works in both directions with tab or shift+tab. No more focus tab on background.
Refs: #738, #3689

### CODE REVIEW

**Comment by wesleycho** [comment]
Does this work with multiple modals?

**Comment by paulorbpacheco** [comment]
I didn't use with multiple modals, I tested using the main components page. I will create a plunk today and post here.

**Comment by paulorbpacheco** [comment]
http://plnkr.co/izQCgs71XzDt3anp7ZH7

**Comment by e-cloud** [comment]
nice, :+1:

File: src/modal/docs/demo.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h3 class="modal-title">I'm a modal!</h3>
         <div class="modal-body">
             <ul>
                 <li ng-repeat="item in items">
-                    <a ng-click="selected.item = item">{{ item }}</a>
+                    <a href="#" ng-click="selected.item = item">{{ item }}</a>
```
Comment (wesleycho) [inline src/modal/docs/demo.html]: Any reason for this change? I think you should revert this, as I think it would have an adverse effect on the docs page.

Comment (wesleycho) [inline src/modal/docs/demo.html]: Hmm, I think I get it - maybe change ng-click to
```
$event.preventDefault(); selected.item = item
```

**Comment by wesleycho** [comment]
Work here is pretty solid, just needs some code style clean up - I will merge this shortly with all of the needed minor/cosmetic fixes.

**Comment by e-cloud** [comment]
@wesleycho , @paulorbpacheco
**thanks for your effort** on the **traping tab feature on the modal module**. but i have a trouble with the new feature.

i use a tabset in the modal. when i update to **v0.13.2**, my modals without tabs work as expected, but it turn to tabs and modal aggregation, it can only trap the `tab` key on the second tab of the tabset.

here is a plukr that reproduce the problem.
http://plnkr.co/edit/wwgZhXu0fru4L703z5by?p=preview

i'm not sure if i use it on the wrong place or the modal source code has some bugs not concerned yet.
can you guys offset some help?

**Comment by wesleycho** [comment]
Can you file a new issue? This isn't the place to discuss it.

**Comment by e-cloud** [comment]
ok, i have filed it just now

**Comment by paulorbpacheco** [comment]
Hello @e-cloud,
Thank you so much for find this bug. I would never think to use tabs inside modals :). Did you create the issue?

**Comment by wesleycho** [comment]
#4149

**Pull request closed without merging**